graylog2-server/src/main/java/org/graylog2/indexer/NodeInfoCache.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.indexer;
+
+import com.github.joschi.jadconfig.util.Duration;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import org.graylog2.indexer.cluster.Cluster;
+import org.graylog2.indexer.guava.FunctionCacheLoader;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Optional;
+
+@Singleton
+public class NodeInfoCache {
+    private static final Duration EXPIRE_DURATION = Duration.minutes(1L);
+    private final LoadingCache<String, Optional<String>> nodeNameCache;
+    private final LoadingCache<String, Optional<String>> hostNameCache;
+
+    @Inject
+    public NodeInfoCache(Cluster cluster) {
+        this(new FunctionCacheLoader<>(cluster::nodeIdToName), new FunctionCacheLoader<>(cluster::nodeIdToHostName));
+    }
+
+    private NodeInfoCache(CacheLoader<String, Optional<String>> nodeNameCache,
```

Comment:
✅ The intention was to use them in test cases but since they aren't used in that way, I'll remove them.